src/main/java/org/towerhawk/monitor/check/type/http/HttpCheck.java

```diff
@@ -0,0 +1,116 @@
+import org.towerhawk.monitor.check.impl.AbstractCheck;
+import org.towerhawk.monitor.check.run.CheckRun;
+import org.towerhawk.monitor.check.run.context.RunContext;
+import org.towerhawk.monitor.check.threshold.Threshold;
+import org.towerhawk.serde.resolver.CheckType;
+import org.towerhawk.spring.config.Configuration;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.nio.charset.StandardCharsets;
+
+@Slf4j
+@CheckType("http")
+@Setter
+public class HttpCheck extends AbstractCheck {
+	protected String endpoint;
+	protected String method = "GET";
+	protected String body;
+	protected Auth auth = new Auth();
+	protected boolean includeResponseInContext = true;
+
+	protected HttpComponentsClientHttpRequestFactory requestFactory;
+
+	protected CloseableHttpClient configureHttpClient() {
+		HttpClientBuilder builder = HttpClients.custom();
+
+		if ( StringUtils.hasText(auth.getUsername()) ) {
+			Credentials creds = new UsernamePasswordCredentials(auth.getUsername(), auth.getPassword());
+			CredentialsProvider provider = new BasicCredentialsProvider();
+			provider.setCredentials(AuthScope.ANY, creds);
+			builder.setDefaultCredentialsProvider(provider);
+		}
+
+		return builder.build();
+	}
+
+	protected ClientHttpRequest getClientHttpRequest() {
+		try {
+			return requestFactory.createRequest(new URI(endpoint), HttpMethod.resolve(method));
+		} catch (IOException | URISyntaxException e) {
+			throw new IllegalArgumentException(e);
+		}
+	}
+
+	@Override
+	protected void doRun(CheckRun.Builder builder, RunContext context) throws InterruptedException {
+		try (ClientHttpResponse response = getClientHttpRequest().execute(); ){
+			Threshold t = getThreshold();
+			String asString = IOUtils.toString(response.getBody(), StandardCharsets.UTF_8);
+			if ( includeResponseInContext ) {
+				context.putContext("org.towerhawk.monitor.check.type.http.httpResponse", asString);
+			}
+			CheckRun.Status status = t.evaluate(builder, asString);
+			if ( status == CheckRun.Status.SUCCEEDED ) {
+				builder.succeeded().addContext("connection", String.format("Connection to %s successful", endpoint));
+			}
+		} catch ( IOException e ) {
+			throw new IllegalStateException(e);
+		}
+	}
+
+	@Override
+	public void init(Check check, Configuration configuration, App app, String id) {
+		if ( !StringUtils.hasText(endpoint) ) {
+			endpoint = new StringBuilder("http://").append(configuration.getDefaultHost()).toString();
+		}
+
+		HttpClient client = configureHttpClient();
+		requestFactory = new HttpComponentsClientHttpRequestFactory(client) {
+			@Override
+			protected void postProcessHttpRequest(HttpUriRequest request) {
+				if ( request instanceof HttpEntityEnclosingRequest) {
+					((HttpEntityEnclosingRequest) request).setEntity(new ByteArrayEntity(body.getBytes(StandardCharsets.UTF_8)));
+				}
+			}
+		};
+
+		super.init(check, configuration, app, id);
+	}
+
+	public class Auth {
+		@Getter
```

Comment:
You can add @Getter and @Setter on the class to get this same functionality as well